src/logger.ts

```diff
@@ -84,33 +90,51 @@ export class Logger {
 
     private getPrefixesFor(command: Command): Record<string, string> {
         return {
-            pid: String(command.pid),
+            pid: command.pid != null ? String(command.pid) : '',
```

Comment:
So that it's not `'undefined'` when there's limited concurrency, which could be longer than the actual PID

Reply:
```suggestion
            // When there's limited concurrency, the PID might not be immediately available,
            // so avoid the string 'undefined' from becoming a prefix
            pid: command.pid != null ? String(command.pid) : '',
```